src/App.jsx

```diff
@@ -1,17 +1,71 @@
+import React from 'react';
 import './App.css';
+import { Route, Switch, BrowserRouter } from 'react-router-dom';
+import FilterByDuration from './components/filterByDuration/FilterByDuration';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <h1>React starter template</h1>
-        <p>Since React 17, you don&#39;t need this anymore:</p>
-        <pre>
-          <code>import React from &#39;react&#39;;</code>
-        </pre>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startTime: '20:00',
+      endTime: '22:00',
+      runtime: 120,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    // this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  /**
+   * This method allows you to calculate the maximum duration of a film which will
+   * be used to filter during the request to the API
+   */
+  calculruntime() {
+    const { startTime, endTime } = this.state;
+    // calculation of the maximum length of the film
+    const xstart =
+      Number(startTime.substr(0, 2)) * 60 + Number(startTime.substr(3));
+    const xend = Number(endTime.substr(0, 2)) * 60 + Number(endTime.substr(3));
+    if (xstart > xend) {
+      this.setState({
+        runtime: 1440 - xstart + xend,
+      });
+    }
+    this.setState({
+      runtime: xend - xstart,
+    });
+  }
+
+  handleChange(event) {
+    const { name } = event.target;
+    this.setState(
+      {
+        [name]: event.target.value,
+      },
+      this.calculruntime,
+    );
+  }
+
+  render() {
+    const { startTime, endTime, runtime } = this.state;
+    return (
+      <div className="App">
+        <h1>entre amis</h1>
```

Comment:
I think <h1> should be in the common header, so this title should be in <h2> tags

Reply:
OK for the next sprint